test/python/transpiler/test_unroller.py

```diff
@@ -297,6 +297,155 @@ def test_unrolling_global_phase_nested_gates(self):
 
         self.assertEqual(Operator(qc), Operator(qcd))
 
+    def test_if_simple(self):
+        """Test a simple if statement unrolls correctly."""
+        qubits = [Qubit(), Qubit()]
+        clbits = [Clbit(), Clbit()]
+
+        qc = QuantumCircuit(qubits, clbits)
+        qc.h(0)
+        qc.measure(0, 0)
+        with qc.if_test((clbits[0], 0)):
+            qc.x(0)
+        qc.h(0)
+        qc.measure(0, 1)
+        with qc.if_test((clbits[1], 0)):
+            qc.h(1)
+            qc.cx(1, 0)
+        dag = circuit_to_dag(qc)
+        unrolled_dag = Unroller(["u", "cx"]).run(dag)
+        unrolled_circ = dag_to_circuit(unrolled_dag)
+
+        expected = QuantumCircuit(qubits, clbits)
+        expected.u(pi / 2, 0, pi, 0)
+        expected.measure(0, 0)
+        with expected.if_test((clbits[0], 0)):
+            expected.u(pi, 0, pi, 0)
+        expected.u(pi / 2, 0, pi, 0)
+        expected.measure(0, 1)
+        with expected.if_test((clbits[1], 0)):
+            expected.u(pi / 2, 0, pi, 1)
+            expected.cx(1, 0)
+
+        for ic1, ic2 in zip(unrolled_circ.data, expected.data):
+            inst1, inst2 = ic1[0], ic2[0]
+            if isinstance(inst1, ControlFlowOp):
+                for param1, param2 in zip(inst1.params, inst2.params):
+                    if isinstance(param1, QuantumCircuit):
+                        for ic3, ic4 in zip(param1.data, param2.data):
+                            self.assertEqual(ic3[0], ic4[0])
```

Comment:
Is there a reason not to do `self.assertEqual(unrolled_circ, expected)`?  I ran 1,000 tests for each of the cases here, and didn't see any issues.